src/Servers/Kestrel/test/SystemdActivation/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet-nightly:2.0-runtime-deps
+FROM mcr.microsoft.com/dotnet/runtime-deps:2.1
```

Comment:
Should we instead be using the `6.0` or `7.0.0-preview.5` tag in this branch❔

Reply:
I'd vote for the 7.0-preview5 tag

Reply:
Are you also suggesting we use the `3.1` or perhaps `3.1-focal` tag in release/3.1 and `6.0` or `6.0-focal` in release/6.0❔ At the moment, the src/Servers/Kestrel/test/SystemdActivation/Dockerfile files are identical in all existing branches.

The same is _almost_ true of src/Servers/Kestrel/samples/Http2SampleApp/Dockerfile: The containing app's TFM was changed to `netcoreapp2.2` sometime between 2.1 and 3.1 and c5c8f4385ee8 removed Docker bits from it in 6.0. There, the choices are between backporting @BrennanConroy's commit, moving to newer tags (including newer Ubuntu or Debian base images), or leaving #42245 and #42246 alone (modulo the above paragraph).

Reply:
Yeah, as long as we're moving from 2.0 everywhere to 2.1 everywhere, I think it makes sense to align on TFM. I think we can leave the sample alone in the other 2 PRs, I'm not too concerned with samples in servicing branches being perfect.

Reply:
@halter73 @Tratcher @BrennanConroy @JamesNK what are your thoughts on these two Docker files❔ Neither is used and neither has been kept up-to-date.

Reply:
I believe it's used with https://github.com/dotnet/aspnetcore/tree/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Servers/Kestrel/samples/SystemdTestApp when doing local debugging/testing but likely hasn't actually been used in years 😄 
I would assume it would need whatever TFM the branch is running on.